app.py

```
@@ -1,36 +1,79 @@
+            start_new_session=True  # This creates a new process group
+        )
+    
+    # Register function to terminate worker when app exits
+    def cleanup():
+        print("Shutting down RQ worker...")
+        if sys.platform == "win32":
+            worker_process.send_signal(signal.CTRL_BREAK_EVENT)
+        else:
+            try:
+                os.killpg(os.getpgid(worker_process.pid), signal.SIGTERM)
+            except:
```

Ruff (0.8.2)

Do not use bare `except`

(E722)